pymc_marketing/mmm/causal.py

```diff
@@ -34,6 +39,949 @@ def __init__(self, *args, **kwargs):
+
+        \mathrm{BIC}(M) = n \log\!\left(\frac{\mathrm{RSS}}{n}\right)
+                          + k \log(n),
+
+    with residual sum of squares :math:`\mathrm{RSS}`, sample size :math:`n`,
+    and number of parameters :math:`k`.
+
+    The Bayes factor is approximated by
+
+    .. math::
+
+        \log \mathrm{BF}_{10} \approx -\tfrac{1}{2}
+        \left[ \mathrm{BIC}(M_1) - \mathrm{BIC}(M_0) \right].
+
+    Independence is declared if :math:`\mathrm{BF}_{10} < \tau`,
+    where :math:`\tau` is set via the ``bf_thresh`` parameter.
+
+    Target Edge Rules
+    -----------------
+    Different rules govern how driver → target edges are retained:
+
+    - ``"any"``:
+      keep :math:`X \to Y` unless **any** conditioning set renders
+      :math:`X \perp Y \mid S`.
+    - ``"conservative"``:
+      keep :math:`X \to Y` if **at least one** conditioning set shows
+      dependence.
+    - ``"fullS"``:
+      test only with the **full set** of other drivers as :math:`S`.
+
+    Examples
+    --------
+    **1. Basic usage with full conditioning set**
+
+    .. code-block:: python
+
+        import numpy as np, pandas as pd
+
+        rng = np.random.default_rng(7)
+        n = 2000
+        C = rng.gamma(2,1,n)
+        A = 0.7*C + rng.gamma(2,1,n)
+        D = 0.5*C + rng.gamma(2,1,n)
+        B = 0.8*A + rng.gamma(2,1,n)
+        Y = 0.9*B + 0.6*D + 0.7*C + rng.gamma(2,1,n)
+
+        df = pd.DataFrame({"A":A,"B":B,"C":C,"D":D,"Y":Y})
+        df = (df - df.mean())/df.std()  # recommended scaling
+
+        model = TBFPC(target="Y", target_edge_rule="fullS")
+        model.fit(df, drivers=["A","B","C","D"])
+
+        print(model.get_directed_edges())
+        print(model.get_undirected_edges())
+        print(model.to_digraph())
+
+    **2. Using forbidden edges**
+
+    You can specify edges that must *not* be tested or included
+    (prior knowledge about the domain).
+
+    .. code-block:: python
+
+        model = TBFPC(
+            target="Y",
+            target_edge_rule="any",
+            forbidden_edges=[("A","C")]  # forbid A--C
+        )
+        model.fit(df, drivers=["A","B","C","D"])
+        print(model.to_digraph())
+
+    **3. Conservative rule**
+
+    Keeps driver → target edges if **any conditioning set**
+    shows dependence.
+
+    .. code-block:: python
+
+        model = TBFPC(target="Y", target_edge_rule="conservative")
+        model.fit(df, drivers=["A","B","C","D"])
+        print(model.to_digraph())
+
+    References
+    ----------
+    - Spirtes, Glymour, Scheines (2000). *Causation, Prediction, and Search*. MIT Press. [PC algorithm]
+    - Spirtes & Glymour (1991). "An Algorithm for Fast Recovery of Sparse Causal Graphs."
+    - Kass, R. & Raftery, A. (1995). "Bayes Factors."
+    """
+
+    def __init__(
+        self,
+        target: str,
+        *,
+        target_edge_rule: str = "any",
+        bf_thresh: float = 1.0,
+        forbidden_edges: Sequence[tuple[str, str]] | None = None,
+    ):
```

Comment:
can we add doctrings for this init method? Also, can we use Pydantic to validate the call and use Field ensuting `target` is within a lsit of possible values (say with `Literal` type)?